src/Conversion/KrnlToLLVM/KrnlInstrument.cpp

```diff
@@ -61,7 +61,28 @@ class KrnlInstrumentOpLowering : public ConversionPattern {
     StringRef nodeName;
     if (instrumentOp.getNodeName().has_value())
       nodeName = instrumentOp.getNodeName().value();
-    else
+    else if (auto nameLoc = loc.dyn_cast<NameLoc>())
+      nodeName = nameLoc.getName();
+    else if (auto fusedLoc = loc.dyn_cast<FusedLoc>()) {
+      // Combine each location name and set it as nodeName.
+      std::string name;
+      for (Location locIt : fusedLoc.getLocations()) {
+        if (auto nameLocIt = locIt.dyn_cast<NameLoc>())
+          name += nameLocIt.getName().str() + "-";
+        else if (auto fileLineColLoc = locIt.dyn_cast<FileLineColLoc>())
+          name += fileLineColLoc.getFilename().str() + "-";
+      }
+      if (name.empty())
+        name = "NOTSET";
+      name.pop_back(); // remove last "-"
```

Comment:
This will erase "T" in case of "NOTSET". Should be covered by `else`?

Reply:
Oops. Thanks for the comment.